distributed/scheduler.py

```diff
@@ -1522,6 +1529,10 @@ def group_key(self):
     def prefix_key(self):
         return self._prefix._name
 
+    @property
+    def erred_on(self):
+        return self._erred_on
```

Comment:
One of the reasons why the erred task state is never forgotten is that keys are only "remembered" in `processing_on (processing)` or `who_has (memory)`, depending on the state of the task. This would be the equivalent for erred tasks to allow us to tell the worker to forget the task.

@jakirkham you have been involved in the scheduler state machine a lot recently. Just pinging in case you have thoughts about adding more state here or if you see other options